### PR TITLE
[DevEx] Switched circleci-heroku docker image to use cimg/python:3.7 as base

### PR DESCRIPTION
**Pull request by ecarlste**
We are starting to run some of our python helper scripts in the `heroku_docker/build_deploy_release` CircleCI jobs, and we will be using more and more of these to help with other tasks like sending logs to sumo during steps where we use timing. Eventually, we'll want to install the tools using something like `pipenv` on the image itself. For now, until we have something in place to auto-update the images when the code is updated, we'll be doing that in the CircleCI steps.

### CODE REVIEW

File: circleci-heroku/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,3 @@
-FROM cimg/base:stable
+FROM cimg/python:3.7
```
Comment (groves) [inline circleci-heroku/Dockerfile:1]: Does python:3.7 build on top of base:stable?

**Review by groves — APPROVED** [review]

File: circleci-heroku/Dockerfile (hunk 1 above)
Comment (ecarlste) [reply circleci-heroku/Dockerfile:1]: I'm not sure if it builds on top of stable or not, I'll check the Dockerfile for it.

Comment (ecarlste) [reply circleci-heroku/Dockerfile:1]: it doesn't, it builds off `cimg/base:2020.12` shown here: https://github.com/CircleCI-Public/cimg-python/blob/master/3.7/Dockerfile

Comment (groves) [reply circleci-heroku/Dockerfile:1]: Sounds like that pulls off of stable at that point in time, so should be fine: https://github.com/CircleCI-Public/cimg-base#tagging-scheme

**Pull request merged**